### PR TITLE
Added composite hydrator naming strategy

### PR DESCRIPTION
**Pull request by ojhaujjwal**
This naming strategy allows us to use different naming strategy for different property.

### CODE REVIEW

File: library/Zend/Stdlib/Hydrator/NamingStrategy/CompositeNamingStrategy.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline library/Zend/Stdlib/Hydrator/NamingStrategy/CompositeNamingStrategy.php]: 2015 ;)

File: tests/ZendTest/Stdlib/Hydrator/NamingStrategy/CompositeNamingStrategyTest.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2014 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (samsonasik) [inline tests/ZendTest/Stdlib/Hydrator/NamingStrategy/CompositeNamingStrategyTest.php]: 2015 ;)

**Comment by ojhaujjwal** [comment]
@Ocramius Done!

**Comment by Ocramius** [comment]
@ojhaujjwal merged, thanks! Please note that I also introduced a new [`IdentityNamingStrategy`](https://github.com/zendframework/zf2/blob/688f5c165d24011ddc5d38b0caac8ec9f091b721/library/Zend/Stdlib/Hydrator/NamingStrategy/IdentityNamingStrategy.php) to simplify this logic on merge.

`develop` zendframework/zf2@688f5c165d24011ddc5d38b0caac8ec9f091b721

**Comment by ojhaujjwal** [comment]
@Ocramius  :+1:

**Pull request closed without merging**